testnet/stacks-node/src/config.rs

```diff
@@ -1620,18 +1633,21 @@ impl BurnchainConfigFile {
                     // Using std::net::LookupHost would be preferable, but it's
                     // unfortunately unstable at this point.
                     // https://doc.rust-lang.org/1.6.0/std/net/struct.LookupHost.html
-                    let mut sock_addrs = format!("{}:1", &peer_host)
-                        .to_socket_addrs()
-                        .map_err(|e| format!("Invalid burnchain.peer_host: {}", &e))?;
-                    let sock_addr = match sock_addrs.next() {
-                        Some(addr) => addr,
-                        None => {
+                    let mut attempts = 0;
+                    let mut addrs_iter = loop {
+                        if let Ok(addrs_iter) = format!("{peer_host}:1").to_socket_addrs() {
+                            break addrs_iter;
+                        }
+                        attempts += 1;
+                        if attempts == 15 {
                             return Err(format!(
                                 "No IP address could be queried for '{}'",
                                 &peer_host
                             ));
                         }
+                        sleep(std::time::Duration::from_secs(5));
                     };
+                    let sock_addr = addrs_iter.next().expect("Failed to resolve IP address");
```

Comment:
Let's make this return an error instead of an `expect`.

Reply:
This is no longer part of this PR. It's part of @hugocaillard's PR, I'll tag him so he can see this

https://github.com/stacks-network/stacks-core/pull/4234/files#diff-8c19dd62a8d676279eb950faf846e62e1833a2fc4430aa07a4ae47229b2a2c21R1617